addons/product_expiry/models/production_lot.py

```diff
@@ -15,6 +15,17 @@ class StockProductionLot(models.Model):
         help='This is the date on which the goods with this Serial Number should be removed from the stock.')
     alert_date = fields.Datetime(string='Alert Date',
         help="This is the date on which an alert should be notified about the goods with this Serial Number.")
+    product_expiry_alert = fields.Char(compute='_compute_product_expiry_alert')
+
+    @api.depends('alert_date', 'product_id')
+    def _compute_product_expiry_alert(self):
+        current_date = fields.Date.from_string(fields.Date.today())
+        for lot in self.filtered(lambda l: l.alert_date and l.product_id):
+            date_alert = fields.Date.from_string(lot.alert_date)
+            if date_alert == current_date:
+                lot.product_expiry_alert = _('Expiration Alert')
+            elif date_alert < current_date:
+                lot.product_expiry_alert = _('Expiration Date Passed')
```

Comment:
Don't name it 'Expiration date passed' because it might not be the case. The alert date might be different than the "best before date" or the "removal date".

Call it Expiration Alert.